### PR TITLE
Future proof the operation's ID

### PR DESCRIPTION
**Pull request by MichaelMure**
- Fix a potential problem where en operation's ID would change if go's json code changed
- rename Hash to ID to be consistent
- ID() doesn't return an error anymore

I'm still pondering if a custom type should be used instead of `string` (kind of like how `git.Hash` was used, but without any relation to git).

### CODE REVIEW

**Comment by MichaelMure** [comment]
Also note: this is not a breaking change :tada:

File: bridge/gitlab/import.go [hunk 1]
```diff
@@ -185,10 +184,10 @@ func (gi *gitlabImporter) ensureNote(repo *cache.RepoCache, b *cache.BugCache, n
 			_, err = b.EditCommentRaw(
```
Comment (A-Hilaly) [inline bridge/gitlab/import.go]: `errResolve` here can never equal `cache.ErrNoMatchingOp` you think we should drop the first error check ?

**Comment by A-Hilaly** [comment]
> I'm still pondering if a custom type should be used instead of string (kind of like how git.Hash was used, but without any relation to git).

I think it is good to differentiate between normal ID's and git hashes, also`git.Hash` got `String()` method. This make sense for me

File: bridge/gitlab/import.go [hunk 2]
```diff
@@ -130,10 +129,10 @@ func (gi *gitlabImporter) ensureIssue(repo *cache.RepoCache, issue *gitlab.Issue
 }
 
 func (gi *gitlabImporter) ensureNote(repo *cache.RepoCache, b *cache.BugCache, note *gitlab.Note) error {
-	id := parseID(note.ID)
+	gitlabID := parseID(note.ID)
 
-	hash, errResolve := b.ResolveOperationWithMetadata(keyGitlabId, id)
-	if errResolve != nil && errResolve != cache.ErrNoMatchingOp {
```
[the remark this answers is not in the document]
Comment (MichaelMure) [reply bridge/gitlab/import.go]: huu, actually I'm not sure why I made this change anymore. Does that make sense to you ?

File: graphql/resolvers/operations.go [hunk 3]
```diff
@@ -6,39 +6,74 @@ import (
...
+var _ graph.EditCommentOperationResolver = editCommentOperationResolver{}
+
 type editCommentOperationResolver struct{}
 
+func (editCommentOperationResolver) ID(ctx context.Context, obj *bug.EditCommentOperation) (string, error) {
+	return obj.Id().String(), nil
+}
+
+func (editCommentOperationResolver) Target(ctx context.Context, obj *bug.EditCommentOperation) (string, error) {
+	panic("implement me")
```
Comment (MichaelMure) [inline graphql/resolvers/operations.go]: @A-Hilaly  :-O

File: bridge/gitlab/import.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (A-Hilaly) [reply bridge/gitlab/import.go]: Thinking a a little bit now i don't think it's correct to do this. We shouldn't trust `nil` errors and return them like in Github importer. like a i said above having an imported operation doesn't mean we have the latest error

File: graphql/resolvers/operations.go (hunk 3 above)
Comment (A-Hilaly) [reply graphql/resolvers/operations.go]: Who did this x) ?? Is it complex to implement ?

**Pull request merged**